File: javascript/leetcode/520. Detect Capital.js
/**
 * @param {string} word
 * @return {boolean}
 */
var detectCapitalUse = function (word) {
  let count = 0;

  for (let ch of word) {
    if (ch == ch.toUpperCase()) ++count;
  }

  return (
    count === word.length ||
    count === 0 ||
    (count === 1 && word[0] === word[0].toUpperCase())
  );
};

File: c++/leetcode/520. Detect Capital.cpp
class Solution
{
public:
  bool detectCapitalUse(string word)
  {
    int count{};

    for (auto ch : word)
    {
      if (isupper(ch))
        ++count;
    }

    return count == word.size() || count == 0 || count == 1 && isupper(word[0]);
  }
};